nitorch/tools/preproc.py

```diff
@@ -24,17 +21,58 @@
 from .spm import affine as apply_affine
 
 
-__all__ = ['load_3d', 'reslice2world', 'reset_origin', 'write_img']
+def get_origin_mat(dim, vx=None, dtype=torch.float64, device='cpu'):
```

Comment:
Quite similar to `nitorch.spatial.affine_default` (it's very recent)

Reply:
Okay, I will keep the one I am using for now (it is used in tools.preproc too).

But I think I should definitely consolidate to make sure there are no replicate functions anywhere.

It seems as if your affine library has a lot of stuff that is in the tools folder, and I think you also have a new implementation of finite differences?

Reply:
Yep, I'll open an issue so that we remember to merge duplicated functions

Reply:
You can actually open an issue directly from your comment.